bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/utils/DynamicDeckUtils.java

```diff
@@ -0,0 +1,340 @@
+            return;
+        }
+        Resource assetResource = master.adaptTo(Resource.class);
+        if (assetResource != null && !ResourceUtil.isNonExistingResource(assetResource)) {
+            Resource content = assetResource.getChild(JcrConstants.JCR_CONTENT);
+            if (content != null && !ResourceUtil.isNonExistingResource(content)
+                    && content.adaptTo(ModifiableValueMap.class) != null) {
+                ModifiableValueMap contentProperties = content.adaptTo(ModifiableValueMap.class);
+                if (contentProperties != null) {
+                    contentProperties.put(DynamicDeckDynamoIDSConstants.IDS_EXPORTED, true);
+                    contentProperties.put(DamConstants.DAM_ASSET_STATE, DamConstants.DAM_ASSET_STATE_PROCESSING);
+                }
+            }
+        }
+    }
+
+    public static Map<String, String> getCollectionsListForLoggedInUser(String collectionQuery, ResourceResolver resourceResolver) {
+
+        if (StringUtils.isEmpty(collectionQuery)) {
+            LOGGER.error("Collection query string to fetch all collections for logged-in user is null/empty");
+            return Collections.emptyMap();
+        }
+        Map<String, String> collectionMap = new HashMap<>();
+        Iterator<Resource> resultItr = resourceResolver.findResources(collectionQuery, javax.jcr.query.Query.JCR_SQL2);
+        while (resultItr.hasNext()) {
+
+            Resource resultCollectionResource = resultItr.next();
+            LOGGER.debug("CollectionList Query Result Resource:{}", resultCollectionResource);
+
+            if (DynamicDeckDynamoConstants.PN_LIGHTBOX_COLLECTION.equalsIgnoreCase(resultCollectionResource.getName())) {
+                continue;
+            }
+
+            collectionMap.put(resultCollectionResource.getValueMap().get(JcrConstants.JCR_TITLE, String.class), resultCollectionResource.getPath());
+
+        }
+        return collectionMap;
+    }
+
+    /**
+     * This method will return asset Input stream for indd file
+     *
+     * @param resourceResolver
+     * @param assetResource
+     * @return
+     * @throws DynamicDeckDynamoException
+     */
+    public static InputStream getInddXmlRenditionInputStream(ResourceResolver resourceResolver,
+                                                             Resource assetResource) throws DynamicDeckDynamoException {
+        Resource assetJcrContentResource = resourceResolver.getResource(assetResource.getPath() + DynamicDeckDynamoConstants.SLASH
+                + JcrConstants.JCR_CONTENT + DynamicDeckDynamoConstants.SLASH + DamConstants.RENDITIONS_FOLDER + DynamicDeckDynamoConstants.SLASH
+                + StringUtils.replace(assetResource.getName(), DynamicDeckDynamoConstants.DOT + DynamicDeckDynamoConstants.INDD_EXTENSION,
+                DynamicDeckDynamoConstants.DOT + DynamicDeckDynamoConstants.XML_EXTENSION)
+                + DynamicDeckDynamoConstants.SLASH + JcrConstants.JCR_CONTENT);
```

Comment:
Please split this up and make it more understandable. Maybe using String.format() and format it properly.

Reply:
Would probably also be more readable to just use `Asset.getRendition().getInputStream()` here. Or some other way of using the Asset API "natively".